lib/npm/script/run.js

```diff
@@ -22,7 +22,17 @@ function runScript(context, script, msg, next) {
       return next(err);
     }
     var options = createOptions(context.path, context);
-    var proc = spawn(_path, [], options);
+    var scriptExtension = path.extname(_path);
+    var proc = '';
+    if ((['.bat','.cmd'].indexOf(scriptExtension) !== -1 && process.platform === 'win32') ||
+     (scriptExtension === '.sh' && process.platform !== 'win32')) {
+      proc = spawn(_path, [], options);
+    } else if (scriptExtension === '.ps1') {
+      proc = spawn('powershell.exe',[_path], options);
+    } else {
+      context.emit('fail', 'verbose', 'unsupported', scriptExtension + ' not supported');
+      return next(Error(msg));
+    }
```

Comment:
@MylesBorins the extension is no longer required, I have updated it to behave the same as before when there is no file path